### ISSUE TITLE
Internal temperature

### ISSUE THREAD

**Issue by dagenest** [tesneddon/wxWmr500#5]
First, thanks a lot for your driver, it works perfectly on my new WMR500 with a selfhosted mosquitto server.
It seems that the "Inside temperature" is wrong: on my station, _indoor['w9']['c93'])_ seems to be the minimal daily temp, whereas **c91** seems to be the actual temperature (and it seems coherent with the external temperature, c3**1**).

**Comment by tesneddon** [comment]
Wow, a lot of interest all of a sudden :-) I'm glad you got it working. Thanks for the input on the data fields. I will update that in the next couple days.

The gust speed and direction are both still wrong. I'm still trying to figure that one out. If you had. Any insight that would also be helpful.

**Comment by dagenest** [comment]
(Sorry for the Off-Topic)
Thanks. Just a quick note since your github is a relevant page when googling about this station. The official app does not work anymore, so the station language cannot be changed. But once a mosquitto_server is installed and configured it is possible to change the language by sending a mqtt "setSettings" command, for example:
`mosquitto_pub -h <MQTT_HOST> -t "enno/out/json/<DEVICE_ID>" -m "{\"command\": \"setSettings\", \"cb3\" : "<NUMBER>", \"id\": \"DEBUG\"}" -d`
Where `<MQTT_HOST>` is obviously the IP of the MQTT broker, `<DEVICE_ID>` is the device id (really?) and `<NUMBER>` is 0 for English, 1 for French, 2 for German, 3 for Italian, 4 for Spanish, 5 for Russian.

**Comment by dckiller51** [comment]
> First, thanks a lot for your driver, it works perfectly on my new WMR500 with a selfhosted mosquitto server.
> It seems that the "Inside temperature" is wrong: on my station, _indoor['w9']['c93'])_ seems to be the minimal daily temp, whereas **c91** seems to be the actual temperature (and it seems coherent with the external temperature, c3**1**).

Can you detail the setup? Not having access to the server I am blocked. No problem to add a mosquito server.

**Comment by dagenest** [comment]
First you need to connect your station to your wifi network using the app (with a 2.4GHz network, not a 5GHz one). After sending your SSID and Wifi key, the app tries to connect to Oregon's servers and obviously it fails, but it does not matter: your station is connected to your wifi network (See the Wifi icon on the station).
The WMR500 tries to connect to a MQTT broker `mqtt.idtlive.com` but it does not work anymore. So you have to configure DNS resolution of your LAN in order to send MQTT queries to *your* MQTT broker. Depending on your router, it may be simple, or impossible. In the latter case, you may install *dnsmasq* on a server in order to resolve mqtt.idtlive.com as the IP of your MQTT broker (and change the DHCP config of your router in order to send your dnsmasq host IP as DNS server).
Reboot your station.
Once this is done, you have to know the id of your station. Type `mosquitto_sub -v -h <MQTT_BROKER_IP> -t "#"` and after a few seconds / minutes, you should see a message from your station, the id is the `deviceId` property.
Install WeeWx, put Tim's driver in the right directory (`/usr/share/weewx/user/` under Debian with WeeWx installed as debian package).
Note that wxwmr500 requires Python 3, and the default Python interpreter in Debian 10 is Python 2.7.
Edit /etc/weewx/weewx.conf : `station_type = wxWmr500` in `[Station]` and
```
[wxWmr500]
    host = <MQTT_BROKER_IP>
    devid = <DEVICE_ID>
    appid = 5568DD91-E97F-4C68-A567-0B94346681E6
    driver = user.wxWmr500
```
Bon courage ;)

**Comment by dagenest** [comment]
One more thing. If your wifi key is long and/or contains special chars, the WMR500 can't connect (for me). I have a "guest" wifi network with a less complex key and it works with this one.
For Tim: my station works for ~24h without any drop, so it doesn't need a frequent reset.

**Comment by dckiller51** [comment]
> mqtt.idtlive.com

Your explanations are clear but I did not succeed. I use a router with Openwrt. 
It is well in: 
DHCP and DNS | General Settings | DNS forwardings | /mqtt.idtlive.com/192.168.2.192
Dedicated mqtt server under debian 10 ip: 192.168.2.192
Wmr500 ip: 192.168.2.106

**Comment by dagenest** [comment]
Does Openwrt show you the WMR500 in the connected devices?
There is a "verbose" mode for mosquitto:
`echo "log_type all" > /etc/mosquitto/conf.d/logall.conf`
Restart mosquitto, and you shoud see "new client connected from..." in `/var/log/mosquitto/mosquitto.log`
Once it is done, you could try to obtain an answer from the station with `mosquitto_pub -h <MQTT_BROKER> -t "enno/out/json/<DEVICEID>" -m "{\"command\": \"getSettingStatus\", \"id\": \"DEBUG\"}" -d` while you have subscribed for publications with `mosquitto_sub -v -h 192.168.0.10 -t "#"`

**Comment by dckiller51** [comment]
> Openwrt vous montre-t-il le WMR500 dans les appareils connectés ?
> Il existe un mode "verbose" pour mosquitto :
> `echo "log_type all" > /etc/mosquitto/conf.d/logall.conf`
> Redémarrez mosquitto, et vous devriez voir "nouveau client connecté depuis..." dans `/var/log/mosquitto/mosquitto.log`
> Une fois cela fait, vous pouvez essayer d'obtenir une réponse de la station avec `mosquitto_pub -h <MQTT_BROKER> -t "enno/out/json/<DEVICEID>" -m "{\"command\": \"getSettingStatus\", \"id\": \"DEBUG\"}" -d`pendant que vous vous êtes abonné aux publications avec`mosquitto_sub -v -h 192.168.0.10 -t "#"`

Yes openwrt shows the wmr500 connected.I think my problem is the redirection from mqtt.idtlive.com to the mqtt broker. I think I don't understand or save this configuration in openwrt.

**Comment by dagenest** [comment]
I'm not an Openwrt user, but "DNS forward" does not seem to be what you need (there is no DNS server on your server).
If your openwrt router is your DNS provider, you could try to add to your /etc/hosts : `192.168.2.192 mqtt.idtlive.com`, or install dnsmasq on your Openwrt?
Tim, sorry for the complete off topic!

**Comment by dckiller51** [comment]
> Je ne suis pas un utilisateur d'Openwrt, mais "DNS forward" ne semble pas être ce dont vous avez besoin (il n'y a pas de serveur DNS sur votre serveur).
> Si votre routeur openwrt est votre fournisseur DNS, vous pouvez essayer d'ajouter à votre /etc/hosts : `192.168.2.192 mqtt.idtlive.com`, ou d'installer dnsmasq sur votre Openwrt?
> Tim, désolé pour le hors sujet complet !

Thanks. 

```
enno/out/json/7cc4ff06-c32b-4199-b6cc-fc520af0df79 {"command": "getChannel1Status", "id": "5568DD91-E97F-4C68-A567-0B94346681E6"}
```

**Comment by tesneddon** [comment]
Thanks @dagenest for your help. I had no idea that the official app had broken. I am putting together a github pages landing page and I'll update with some more setup documentation. There must be more than three people out there trying to make this thing work :-)

**Comment by iv1-infovista** [comment]
> First you need to connect your station to your wifi network using the app (with a 2.4GHz network, not a 5GHz one). After sending your SSID and Wifi key, the app tries to connect to Oregon's servers and obviously it fails, but it does not matter: your station is connected to your wifi network (See the Wifi icon on the station). The WMR500 tries to connect to a MQTT broker `mqtt.idtlive.com` but it does not work anymore. So you have to configure DNS resolution of your LAN in order to send MQTT queries to _your_ MQTT broker. Depending on your router, it may be simple, or impossible. In the latter case, you may install _dnsmasq_ on a server in order to resolve mqtt.idtlive.com as the IP of your MQTT broker (and change the DHCP config of your router in order to send your dnsmasq host IP as DNS server). Reboot your station. Once this is done, you have to know the id of your station. Type `mosquitto_sub -v -h <MQTT_BROKER_IP> -t "#"` and after a few seconds / minutes, you should see a message from your station, the id is the `deviceId` property. Install WeeWx, put Tim's driver in the right directory (`/usr/share/weewx/user/` under Debian with WeeWx installed as debian package). Note that wxwmr500 requires Python 3, and the default Python interpreter in Debian 10 is Python 2.7. Edit /etc/weewx/weewx.conf : `station_type = wxWmr500` in `[Station]` and
> 
> ```
> [wxWmr500]
>     host = <MQTT_BROKER_IP>
>     devid = <DEVICE_ID>
>     appid = 5568DD91-E97F-4C68-A567-0B94346681E6
>     driver = user.wxWmr500
> ```
> 
> Bon courage ;)

Salut
I followded your instructuions, but I am not mastering mosquito, is there a special setup for it i jut put
listener 1883
allow_anonymous true

do i have to put more?

And by reboot the staion, you mean reset ? or just remove batteires for a while ?

**Comment by valval274** [comment]
Hi,
I got this damn station and I found out we can't connect anymore to the router with the application, how can I do ? :(

**Comment by tesneddon** [comment]
You can use weewx to display your weather information using dnsmasq to redirect the DNS queries to your own MQTT server (typically mosquitto) and then use the WxWmr500 driver to feed the data into weewx.

**Comment by epatix** [comment]
Hello, 
is there a way to configure wifi without the app?
The app is not working anymore
Thx

**Comment by czech777** [comment]
[epatix](https://github.com/epatix)
via telnet
https://github.com/pcmihnea/HA_WMR500?ref=https://githubhelp.com
